### PR TITLE
Update documentation, and CMakeLists

### PR DESCRIPTION
**Pull request by nguyenhuy0905**
# Update documentation, and CMakeLists

- **What**:

Fix the Ming-gw64 command-line package download instructions to explicitly use `ucrt` packages.
Add general guide on how to debug, format, and lint your code. Add general guide on how to lint the Markdown docs.
I also made the -fsanitize=leak flag optional, and off by default.

- **Why**:

I realized the "generic" package on Ming-gw64 will download to all the places. So, specifying it's UCRT64 makes everything sit inside the UCRT64 directory.
Sanitization is a performance hit, so I let it be optional.

- **How**:

I updated the command to run inside UCRT64, and added a new section called "How-to" for general guide.
If you have any more questions, please create an Issue. Whatever question it is, will be updated on the "how-to."

- **Misc**:

With this out of the way, we are (almost) ready for doing some actual  code.

- **Tags**:

@qu-ngx @AbstractionHLR @BAOELIETRAN @bebeminhminh

---

Pull request by: Huy Nguyen

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -33,11 +33,16 @@ elseif(CMAKE_BUILD_TYPE STREQUAL Debug)
   message("Building a debug version")
 endif()
 message(${CMAKE_SYSTEM_NAME})
+
+option(SANITIZE "Whether to turn on sanitization flags on Unix" OFF)
 # I tried this flag on Windows and it threw an error, so,
-if(NOT (WIN32 OR MSYS))
+if((NOT (WIN32 OR MSYS)) AND (SANITIZE))
```
Comment (nguyenhuy0905) [inline CMakeLists.txt:39]: Sanitization is quite costly in terms of performance, so only use it if really needed.

File: docs/dev-env.md [hunk 2]
```diff
@@ -117,12 +131,14 @@ sudo usermod -aG docker $USER
...
   - In the "Configure" tab, choose the Clang kit.
-  - The extension should know enough to figure out the rest.
+  - In the "Debug" tab, choose either smoldb or smoldb install.
+    - If you choose smoldb install, remember to run this after every build
 
-> [!NOTE]
-> If CMake throws an error, delete and re-create the `build` directory first.
->
+    ```bash
+    cmake --install build
```
Comment (nguyenhuy0905) [inline docs/dev-env.md:140]: This option pulls the executable to the project root.

File: docs/dev-env.md [hunk 3]
```diff
@@ -132,16 +148,21 @@ sudo usermod -aG docker $USER
...
+  Preferably, configure C/C++ to use lldb-vscode.
+  - You should debug using the executable.
 
 ### For Windows system
 
 > [!NOTE]
 > This part is mainly for building a Windows-native executable.
 > If you just need to see the thing run,
-> opt for the [dev container](#set-up-development-environment) way.
+> opt for the [dev container](#the-lazy-way) way.
```
Comment (nguyenhuy0905) [inline docs/dev-env.md:165]: The old hyperlink is kinda worthless since it just goes to the main header.

File: docs/dev-env.md [hunk 4]
```diff
@@ -173,8 +201,9 @@ sudo usermod -aG docker $USER
   # assuming you're on x86_64
   # if you're on an ARM machine, change x86_64 to aarch64
   # compiler
-  pacman -S git gcc clang ninja cmake \
-  mingw-w64-x86_64-clang-tools-extra
+  pacman -S mingw-w64-ucrt-x86_64-clang \
+  mingw-w64-ucrt-x86_64-ninja mingw-w64-ucrt-x86_64-cmake \
+  mingw-w64-ucrt-x86_64-clang-tools-extra
```
Comment (nguyenhuy0905) [inline docs/dev-env.md:206]: I changed the installs to explicitly specify the UCRT64 version. Otherwise, these would be all over the places.

**Review by qu-ngx — APPROVED** [review]
This looks good enough to me. LGTM! But I'm still gonna ask when I'm stuck on something since I don't create them.

Reviewer: Quang Nguyen

**Comment by nguyenhuy0905** [comment]
You know, CMake isn't too shabby once you know a good amount of flags that can be passed to Clang or GCC :smile:

**Pull request merged**